Data/ByteString.hs

```diff
@@ -2075,7 +2079,9 @@ moduleErrorMsg fun msg = "Data.ByteString." ++ fun ++ ':':' ':msg
 
 -- Find from the end of the string using predicate
 findFromEndUntil :: (Word8 -> Bool) -> ByteString -> Int
-findFromEndUntil f ps@(BS x l)
-  | null ps = 0
-  | f (unsafeLast ps) = l
-  | otherwise = findFromEndUntil f (BS x (l - 1))
+findFromEndUntil f ps@(BS x l) = case unsnoc ps of
+  Nothing     -> 0
+  Just (_, b) ->
+    if f b
+      then l
+      else findFromEndUntil f (BS x (l - 1))
```

Comment:
For readability, maybe use the bytestring init returned by `unsnoc` instead of manually creating the new bytestring?

Reply:
do you mean this part `BS x (l - 1)` ?

Reply:
Yes.